src/commands/BootstrapCommand.js

```diff
@@ -55,12 +55,28 @@ export default class BootstrapCommand extends Command {
       });
 
       async.parallelLimit(batch.map(pkg => done => {
-        async.series([
+        // async actions to bootstrap each package
+        const actions = [
           cb => FileSystemUtilities.mkdirp(pkg.nodeModulesLocation, cb),
           cb => this.installExternalPackages(pkg, cb),
           cb => this.linkDependenciesForPackage(pkg, cb),
           cb => this.runPrepublishForPackage(pkg, cb),
-        ], err => {
+        ];
+        // array of dependencies with binaries
+        const depsWithBin = [];
+        // find matching dependencies with binaries
+        this.packages.forEach(dependency => {
+          // for each matched dependency with binaries
+          if (this.hasMatchingDependency(pkg, dependency) && dependency.bin) {
+            depsWithBin.push(dependency);
+          }
+        });
+        // if package has dependencies with binaries
+        if (depsWithBin.length) {
+          // symlink binaries
+          actions.push(cb => this.symlinkDependencyBinaries(pkg, depsWithBin, cb));
```

Comment:
I think the binaries should be linked _before_ `runPrepublishForPackage`.

A package may have a `prepublish` script that uses a binary from a dependency.